src/main/kotlin/org/radarbase/push/integration/garmin/user/GarminServiceUserRepository.kt

```diff
@@ -1,52 +1,201 @@
 package org.radarbase.push.integration.garmin.user
 
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.ObjectReader
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import io.confluent.common.config.ConfigException
+import okhttp3.*
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
 import org.radarbase.gateway.Config
+import org.radarbase.gateway.GarminConfig
 import org.radarbase.push.integration.common.user.User
+import org.radarbase.push.integration.common.user.Users
+import org.radarcns.exception.TokenException
+import org.radarcns.oauth.OAuth2Client
+import org.slf4j.LoggerFactory
 import java.io.IOException
+import java.net.URL
+import java.time.Duration
 import java.time.Instant
+import java.util.*
 import java.util.stream.Stream
 import javax.ws.rs.NotAuthorizedException
 import javax.ws.rs.core.Context
+import kotlin.collections.HashMap
+
 
 class GarminServiceUserRepository(
-    @Context private val config: Config
+        @Context private val config: Config
 ) : GarminUserRepository(config) {
+    private val garminConfig: GarminConfig = config.pushIntegration.garmin
+    private val client: OkHttpClient? = OkHttpClient()
+    private val cachedCredentials: HashMap<String, OAuth1UserCredentials>? = HashMap<String, OAuth1UserCredentials>()
+    private var nextFetch = MIN_INSTANT
+
+    private var baseUrl: HttpUrl
+    private var timedCachedUsers: List<User?> = ArrayList<User?>()
+
+    private var repositoryClient: OAuth2Client? = null
```

Comment:
If this becomes a `val` and is initialised in the constructor, you don't have to force non-null checks `!!` in the rest of the code.

Reply:
What about adding the `lateinit` because there are some conditional checks before the `repositoryClient` is initialized?

Reply:
Lateinit still requires you to initialize the variable in every branch. Instead you can lift the assignment out of the branch (`val smth = if ... else ...`) or still initialize it in every branch.

Reply:
Oh ok I see. But what about the the `clientId` and `clientSecret` that the initializer needs which is taken from the `GarminConfig`?

Reply:
Example init code:
```kotlin
if (tokenUrl != null && clientId.isEmpty()) {
   throw ConfigException("Client ID for user repository is not set.")
}

repositoryClient = if (tokenUrl != null) {
    OAuth2Client.Builder()
         .credentials(clientId, clientSecret)
         .endpoint(tokenUrl)
         .scopes("SUBJECT.READ", "MEASUREMENT.READ", "SUBJECT.UPDATE")
         .httpClient(client)
         .build()
} else null

basicCredentials = if (tokenUrl == null && clientId != null) {
    Credentials.basic(clientId, clientSecret)
} else null
```

Reply:
Oh okay I see. Thanks